pkg/util/iptables/iptables.go

```diff
@@ -502,16 +502,22 @@ const (
 // Monitor is part of Interface
 func (runner *runner) Monitor(canary Chain, tables []Table, reloadFunc func(), interval time.Duration, stopCh <-chan struct{}) {
 	for {
-		for _, table := range tables {
-			if _, err := runner.EnsureChain(table, canary); err != nil {
-				klog.Warningf("Could not set up iptables canary %s/%s: %v", string(table), string(canary), err)
-				return
+		_ = utilwait.PollImmediateUntil(interval, func() (bool, error) {
+			for _, table := range tables {
+				if _, err := runner.EnsureChain(table, canary); err != nil {
+					klog.Warningf("Could not set up iptables canary %s/%s: %v", string(table), string(canary), err)
+					return false, nil
+				}
 			}
-		}
+			return true, nil
+		}, stopCh)
 
 		// Poll until stopCh is closed or iptables is flushed
 		err := utilwait.PollUntil(interval, func() (bool, error) {
-			if runner.chainExists(tables[0], canary) {
+			if exists, err := runner.chainExists(tables[0], canary); exists {
+				return false, nil
+			} else if isResourceError(err) {
+				klog.Warningf("Could not check for iptables canary %s/%s: %v", string(tables[0]), string(canary), err)
 				return false, nil
 			}
 			klog.V(2).Infof("iptables canary %s/%s deleted", string(tables[0]), string(canary))
```

Comment:
Should we be checking more affirmatively that the error returned matched "No chain/target/match by that name" rather than handling all errors this way?